modules/@apostrophecms/schema/ui/apos/components/AposInputRelationship.vue

```diff
@@ -198,7 +200,11 @@ export default {
       }
     },
     async editRelationship (item) {
-      const result = await apos.modal.execute('AposRelationshipEditor', {
+      const editor = this.field.editor === this.imageRelationshipComponent
+        ? this.imageRelationshipComponent
```

Comment:
Do not hardcode the name. The component name should come from `this.field.editor` in the `modal.execute` call.